### PR TITLE
Fix creating model objects with explicit default values (#108)

### PR DESCRIPTION
**Pull request by ewuerger**
This PR fixes the bug described in #108 but for the creation of all AttributeValues, not BooleanAttributeValue exclusively. In doing so it was revealed that handing over timezone unaware `datetime` objects doesn't lead to the supported format for writing to the xml. An extra test parameter was added where in this case the local timezone was added as tzinfo.

### CODE REVIEW

File: tests/test_reqif.py [hunk 1]
```diff
@@ -403,52 +404,60 @@ def test_created_requirements_are_found_from_both_sides(
...
+        xml: str | None,
     ):
         req = model.by_uuid("79291c33-5147-4543-9398-9077d582576d")
         assert isinstance(req, reqif.Requirement)
 
         assert not req.attributes
-        definition = model.by_uuid(def_uuid)
+        definition = model.by_uuid("682bd51d-5451-4930-a97e-8bfca6c3a127")
         attr = req.attributes.create(type_hint, definition=definition)
+        value_attr = req.attributes.create(type_hint, value=value)
```
[the remark this answers is not in the document]
Comment (ewuerger) [reply tests/test_reqif.py]: Can you have a look at [c85f37c](https://github.com/DSD-DBS/py-capellambse/pull/111/commits/c85f37c980e6ad00cc594c3eb29edd604ce0e4fb)? Is that matching your intention?

File: capellambse/loader/xmltools.py [hunk 2]
```diff
@@ -110,13 +110,15 @@ def __get__(self, obj, objtype=None):
...
+        attribute_exists = xml_element.get(self.attribute) is not None
+        if not self.writable and attribute_exists:
             raise TypeError(
                 f"Cannot set attribute {self.__name__!r} on {type(obj).__name__!r} objects"
             )
 
         if value == self.default:
-            self.__delete__(obj)
+            if attribute_exists:
+                self.__delete__(obj)
```
[the remark this answers is not in the document]
Comment (ewuerger) [reply capellambse/loader/xmltools.py]: Can you have a look at [4dc4253](https://github.com/DSD-DBS/py-capellambse/pull/111/commits/4dc4253ac850e4e5e48e1e33085c7748dd9500f2)? Is that matching your intention?

File: tests/test_reqif.py [hunk 3]
```diff
@@ -403,70 +402,86 @@ def test_created_requirements_are_found_from_both_sides(
...
+        assert isinstance(req, reqif.Requirement)
         assert not req.attributes
-        definition = model.by_uuid(def_uuid)
+
+        definition = model.by_uuid("682bd51d-5451-4930-a97e-8bfca6c3a127")
         attr = req.attributes.create(type_hint, definition=definition)
 
-        assert len(req.attributes) == 1
-        assert req.attributes[0] == attr
+        assert tuple(req.attributes) == (attr,)
```
Comment (Wuestengecko) [inline tests/test_reqif.py:436]: We can simplify this even further to `assert req.attributes == [attr]`.

File: tests/test_reqif.py [hunk 4]
```diff
@@ -403,70 +402,86 @@ def test_created_requirements_are_found_from_both_sides(
...
+        assert not req.attributes
 
-        assert len(definitions) + 2 == len(reqtype.attribute_definitions)
-        assert attr_def in reqtype.attribute_definitions
-        assert enum_def in reqtype.attribute_definitions
+        value_attr = req.attributes.create(type_hint, value=value)
+        if isinstance(value, datetime.datetime):
+            value = value.astimezone()
+
+        assert tuple(req.attributes) == (value_attr,)
```
Comment (Wuestengecko) [inline tests/test_reqif.py:482]: Same here, can also be simplified to comparing the list directly.

File: tests/test_reqif.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wuestengecko) [reply tests/test_reqif.py]: Wonderful!

File: capellambse/loader/xmltools.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wuestengecko) [reply capellambse/loader/xmltools.py]: > Didn't know that `self.__delete__` doesn't need to raise an exception.

I wasn't aware of that either before, so don't worry :)

> Can you have a look at [4dc4253](https://github.com/DSD-DBS/py-capellambse/pull/111/commits/4dc4253ac850e4e5e48e1e33085c7748dd9500f2)? Is that matching your intention?

Yes, that looks much cleaner. Well done, young padawan.

**Review by Wuestengecko — CHANGES_REQUESTED** [review]
Just one more minor nitpick, then I'd say it's good to merge.

**Review by Wuestengecko — APPROVED** [review]

**Pull request merged**